proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
```

Comment:
on Wed Jul 06 2016, Andrew Trick <notifications-AT-github.com> wrote:

> I like this:
>
> func initialize<T>(as: T.Type, at: Int = 0, count: Int = 1, to value: T) -> UnsafeMutablePointer<T>
>
> address.initialize(as: String.self, to: "hello")

Changing the order would be more idiomatic, since we already have 

         "hello" as String

  address.initialize(to: "hello", as: String.self)

> 
> address.initialize(as: String.self, at: 3, to: "hello")

That's non-grammatical, and I still don't know whether the `at:`
parameter is measured in bytes or in `MemoryLayout<String>.stride`s.
Which is it?  I can't even make a good recommendation until I know your
intentention.

> address.initialize(as: String.self, count: 3, to: "hello")

I would prefer `count:` to go last and be defaulted to 1

> address.initialize(as: String.self, at: 3, count: 3, to: "hello")

Are all these combinations really needed?  I'm just not sure it's worth
the API complexity.